src/worker/ingestion/process-event.ts

```diff
@@ -254,7 +254,27 @@ export class EventsProcessor {
             )
         }
 
-        let updatedProperties: Properties = { ...propertiesOnce, ...personFound.properties, ...properties }
+        // Figure out which properties we are actually setting
+        const returnedProps: Properties = {}
+        let updatedProperties: Properties = {}
```

Comment:
The `returnedProps` logic is solid, but `updatedProperties` should stay how it was and contain all the properties, not just the changed ones. Otherwise `arePersonsEqualExcludingIncrement` will always be false below and we will run into data loss :party_parrot: